posthog/hogql_queries/hogql_query_runner.py

```diff
@@ -52,6 +62,9 @@ def calculate(self) -> HogQLQueryResponse:
             limit_context=self.limit_context,
             explain=bool(self.query.explain),
         )
+        if paginator:
+            response = response.model_copy(update={**paginator.response_params(), "results": paginator.results})
```

Comment:
Why copy BTW?

Reply:
That's a way I found to update values of the existing `response` Pydantic model.